frame/contracts/rpc/src/lib.rs

```diff
@@ -83,33 +84,47 @@ pub struct CallRequest<AccountId, Balance> {
 	input_data: Bytes,
 }
 
+#[derive(Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+#[serde(rename_all = "camelCase")]
+struct RpcContractExecSuccess {
+	/// The return flags. See `pallet_contracts_primitives::ReturnFlags`.
+	flags: u32,
+	/// Data as returned by the contract.
+	data: Bytes,
+}
+
 /// An RPC serializable result of contract execution
 #[derive(Serialize, Deserialize)]
 #[serde(deny_unknown_fields)]
 #[serde(rename_all = "camelCase")]
-pub enum RpcContractExecResult {
-	/// Successful execution
-	Success {
-		/// The return flags
-		flags: u32,
-		/// Output data
-		data: Bytes,
-		/// How much gas was consumed by the call.
-		gas_consumed: u64,
-	},
-	/// Error execution
-	Error(()),
+pub struct RpcContractExecResult {
+	/// How much gas was consumed by the call. In case of an error this is the amount
+	/// that was used up until the error occurred.
+	gas_consumed: u64,
+	/// Additional dynamic human readable error information for debugging. An empty string
+	/// indicates that no additional information is available.
+	debug_message: String,
```

Comment:
If this is to be used in the future for `debug_println`,  would it be better as `Vec<String>`?

Reply:
This message buffer is meant to be consumed by humans only. I therefore don't see any value in structuring it any further. I want to give the runtime full control over the layout of the message. It won't only consist of lines printed by the contract but also other information that might be helpful in that context.

Even though this buffer is not created in any critical runtime instances (only RPC ones) we still want to minimize allocations when possible. Having one String per line will certainly add more allocations.